.github/workflows/docs.yml

```diff
@@ -35,5 +35,39 @@ jobs:
       - name: Upload docs as GHA artifact
         uses: actions/upload-artifact@v2
         with:
-          name: docs.whl
+          name: built-docs
           path: docs/build/html
+
+  deploy-docs:
+    needs: [build-docs]
+    runs-on: ubuntu-latest
+    if: (github.ref == 'refs/heads/main' && github.event_name == 'push')
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          ref: gh-pages
+          fetch-depth: 3
+
+      - name: Download docs artifact
+        uses: actions/download-artifact@v2
+        with:
+          name: built-docs
+          path: /tmp/docs
+
+      - name: Copy built docs to nightly
+        run: |
+          cp -R /tmp/docs/* nightly/
+          # Manipulations to squash the history:
+          # 1) reset to a fixed commit
+          # 2) use github-actions-x/commit to commit everything as a single commit
+          git reset --soft 218f32faf8b8cc5236bc49ea77ef7d222e21e5d6
```

Comment:
What is this commit?

Reply:
https://github.com/pytorch/functorch/commit/218f32faf8b8cc5236bc49ea77ef7d222e21e5d6

Reply:
If nothing changes on the docs between to commits to pytorch/pytorch, is the commit essentially empty or is there something in it?

If it is essentially empty, then I'm thinking we should comment this out for now and then turn it on later -- I want to be able to manually submit changes to the ghpages branch for the next couple of weeks.

Reply:
This actions is triggered on every commit regardless python docstring or docs/source is unchanged (to make this action aware of specific files changes we can specify that explicitly.)
I think it could be difficult to check at runtime if commit changes or not the docs.
We may opt to another solution using another sub-action: https://github.com/devops-infra/action-commit-push which should allow `--amend` commit option. So, it means when this actions is run and rebuild the docs it would commit ammend the results on gh-pages to the last commit. What do you think ?

Reply:
Hmm, my question is, is the commit "empty" if nothing changes? Does the docs build generate different documentation based on the time of day or does it always generate the same thing if nothing has changed?

I'm trying to evaluate how large of a problem the history will become -- a lot of commits that have very little changes in them (due to us not updating the website) means we don't need to worry about having a lot of commits.